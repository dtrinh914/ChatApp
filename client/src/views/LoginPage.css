.LoginPage{
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
}
.LoginPage form{
    display: flex;
    flex-direction: column;
}

.LoginPage .sign-in{
    margin-bottom: 10px;
}

