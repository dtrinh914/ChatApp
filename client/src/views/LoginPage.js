import React from 'react';
import useInput from '../hooks/useInput';
import {Link, Redirect} from 'react-router-dom';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import './LoginPage.css'

function LoginPage({loggedIn, logInUser}){
    const [username, setUsername] = useInput();
    const [password, setPassword] = useInput();

    const handleSubmit = (e) => {
        e.preventDefault();
        logInUser(username, password);
    }

    if(loggedIn){
        return <Redirect to='/chat' />
    } else {
        return(
            <div className='LoginPage' >
                <form onSubmit={handleSubmit}>
                    <h1>Sign In</h1>
                    <TextField className='sign-in' type='text' name='username' id='username' 
                    label='Username' variant='outlined' value={username} 
                    onChange={setUsername} />
                    <TextField className='sign-in' type="password" name='password' id='password' 
                    label='Password' variant='outlined' value={password} 
                    onChange={setPassword} />
                    <Button type='submit' variant="outlined">Login</Button>
                    <Link to='/users/new'>Create User</Link>
                </form>
            </div>
        );
    }
}

export default LoginPage; 