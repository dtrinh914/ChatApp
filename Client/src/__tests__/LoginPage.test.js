import React from 'react';
import LoginPage from '../views/LoginPage';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);

test('should match snapshot', () => {
    const {asFragment} = render(<BrowserRouter><LoginPage /></BrowserRouter>)
    expect(asFragment()).toMatchSnapshot();
});