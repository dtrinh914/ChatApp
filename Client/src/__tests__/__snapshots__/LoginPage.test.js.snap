// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoginPage"
  >
    <form>
      <h1>
        Sign In
      </h1>
      <label
        for="username"
      >
        Username
      </label>
      <input
        id="username"
        name="username"
        type="text"
        value=""
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        type="password"
        value=""
      />
      <button>
        Login
      </button>
      <a
        href="/users/new"
      >
        Create User
      </a>
    </form>
  </div>
</DocumentFragment>
`;
